extensions/cornerstone/src/services/SyncGroupService/SyncGroupService.ts

```diff
@@ -6,20 +6,36 @@ const EVENTS = {
   TOOL_GROUP_CREATED: 'event::cornerstone::syncgroupservice:toolgroupcreated',
 };
 
+export type SyncCreator = (
```

Comment:
This is the only file with relevant changes - the rest I was working on as I was figuring out how to do this.  Note that this does depend on a few changes in cornerstone 3d to allow synchronization between things other than source/target viewports, but that change is quite small, just in the event delivery mechanism.

Reply:
Now the viewport grid also has a relevant change - but the main changes are in this file, the other one fixes a related bug in viewport grid.